h_program-lang/AST_generic.ml

```diff
@@ -519,6 +519,9 @@ and expr =
       | Range (* .. or ..., Ruby *)
       | Nullish (* ?? in Javascript *)
       | NotNullPostfix (* ! in Typescript, postfix operator *)
+      | LogAndAss (* &&= logical AND assignment, in Javascript *)
```

Comment:
Nope. You're supposed to use instead AssignOp.
There is already And and Or and Nullish in AST_generic.operator type.
Notice that there's no PlusAss, MultAss, ... because all of those are generated by AssignOp (e1, Plus, e2)

Reply:
ok, cool.

Reply:
Actually, they're conditional assignments. The example from the MDN docs below is:

![image](https://user-images.githubusercontent.com/343265/94302398-03254800-ff21-11ea-9701-f8db801c08c3.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment

I don't think we need to add `LogAndAss` but I'll have to make sure I transpile it into the right thing.

Reply:
Does not matter. You can still use AssignOp(e1,And,e2)

Reply:
We dont care about the semantic here

Reply:
Syntactically it is an Assignop And.

Reply:
ok, I did the right thing in semgrep. Closing the PR again.